website/content/en/docs/cloud-providers/AWS/launch-templates.md

```diff
@@ -0,0 +1,233 @@
+
+Alternatively, you may reference AWS's [`bootstrap.sh`
+file](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh)
+when building a custom base image. 
+
+```
+#!/bin/bash
+/etc/eks/bootstrap.sh <my-cluster-name> \
+--kubelet-extra-args <'--max-pods=40'> \
+--b64-cluster-ca <certificateAuthority> \
+--apiserver-endpoint <endpoint> 
+--dns-cluster-ip <serviceIpv4Cidr>
+--use-max-pods false
+```
+
+Note, you must populate this command with live values. Karpenter will
+not change the user data in the launch template. 
+
+Encode using yaml function `!Base64` yaml function or `cat userdata.sh | base64 > userdata-encoded.txt` shell command. 
+
+**Bootstrap Script Parameters**
+
+The sample bootstrap script requires information to join the cluster.
+
+These values may be found using:
+```
+aws eks describe-cluster --name MyKarpenterCluster 
+```
+
+**Kubelet Arguments**
+
+Specifying max-pods can break Karpenter's binpacking logic (it has no way to know what this setting is). If karpenter attempts to pack more than this number of pods, the instance may be oversized, and additional pods will reschedule.
+
+## Situational Fields 
+
+Configure these values in response to a particular use case, such as nodes interacting with another AWS service, or using EBS storage on the node. 
+
+### Instance Profile - IAM
+
+The launch template must include an "instance profile" -- a set of IAM roles. 
+
+The instance profile must include *at least* the permissions of the default Karpenter node instance profile. See the default role, `KarpenterNodeRole`, in the full example below for more information. 
+
+See also, [the managed policy "AmazonEKSWorkerNodePolicy"](https://docs.aws.amazon.com/eks/latest/userguide/security-iam-awsmanpol.html#security-iam-awsmanpol-AmazonEKSWorkerNodePolicy) which includes permission to describe clusters and subnets.
+
+### Storage
+
+Karpenter expects nothing of node storage. Configure as needed for your base
+image.
+
+### Security Groups - Firewall
+
+The launch template must include a security group (i.e., instance firewall rules) and the security group must be associated with the virtual private cloud (VPC) of the EKS cluster.
+
+The security group must permit communication with EKS control plane. Outbound access should be permitted for at least: HTTPS on port 443, DNS (UDP and TCP) on port 53, and your subnet's network access control list (network ACL). 
+
+## Fields with Undefined Behavior
+
+Resources referenced by these fields are controlled by EKS/Karpenter, and not the launch template. 
+
+### Instance Type
+
+The instance type should not be specified in the launch template. Karpenter
+will determine the launch template at run time. 
+
+### Network Interfaces
+
+The [AWS CNI](https://docs.aws.amazon.com/eks/latest/userguide/pod-networking.html) will configure the network interfaces. Do not configure network instances in the launch template.

```

Comment:
Isn't there an alternative to using the CNI? (I can't recall details) Or would we not want to document that?

Reply:
I do not want to document that. If you use a non-standard CNI, it's your responsibilty.